### PR TITLE
feat: experimental reputation client agreement selector

### PR DESCRIPTION
**Pull request by pgrzy-golem**
This feature will allow to choose the best available provider based on benchmarking scores.

### CODE REVIEW

File: src/experimental/reputation/system.ts [hunk 1]
```diff
@@ -41,7 +43,14 @@ export const DEFAULT_AGREEMENT_WEIGHTS: ReputationWeights = {
...
  */
-export const DEFAULT_REPUTATION_URL = "https://reputation.dev-test.golem.network/v1/providers/scores";
+export const DEFAULT_REPUTATION_URL = "https://reputation.dev-test.golem.network/v2/providers/scores";
+
+/**
+ * The number of top scoring providers to consider when selecting an agreement.
+ *
+ * Default for `topPoolSize` agreement selector option.
+ */
+export const DEFAULT_AGREEMENT_TOP_POOL_SIZE = 2;
```
Comment (grisha87) [inline src/experimental/reputation/system.ts:53]: Just curious if that's not to small for the default?

File: src/experimental/reputation/types.ts [hunk 2]
```diff
@@ -25,13 +29,41 @@ export interface ReputationProviderEntry {
   scores: ReputationProviderScores;
 }
 
+/**
+ * Information about a rejected operator.
+ */
+export interface ReputationRejectedOperator {
+  wallet: string;
```
Comment (grisha87) [inline src/experimental/reputation/types.ts:36]: For consistency, please note that we have the `ProviderInfo` type available. It contains `id`, `name` and `WalletAddress`. Perhaps you could use that type here as well.

**Review by grisha87 — APPROVED** [review]

File: src/experimental/reputation/system.ts (hunk 1 above)
Comment (pgrzy-golem) [reply src/experimental/reputation/system.ts:53]: It is small, but also the pool of providers is not big. In my tests keeping this value low did affect the results by alot (10-20%).

Due to the limited amount of good proposals readily available, pool size of 5 had pretty much the same result as default random agreement selector.

**Review by grisha87 — APPROVED** [review]
Looks good, but the build is failing.

**Pull request merged**